### PR TITLE
[RFR] Create/Edit allow changing success message

### PR DESCRIPTION
**Pull request by natrim**
Allow~ changing success message in Create/Edit

### CODE REVIEW

File: docs/CreateEdit.md [hunk 1]
```diff
@@ -177,6 +178,20 @@ const Aside = ({ record }) => (
 
 **Tip**: Always test that the `record` is defined before using it, as react-admin starts rendering the UI before the API call is over.
 
+### Message
```
Comment (fzaninotto) [inline docs/CreateEdit.md]: the name of the prop is too generic. `successMessage` would be better.

File: docs/CreateEdit.md [hunk 2]
```diff
@@ -177,6 +178,20 @@ const Aside = ({ record }) => (
...
+
+By default, on save you will get notification with generic message about change done.
```
Comment (fzaninotto) [inline docs/CreateEdit.md]:
```suggestion
Once the `dataProvider` returns successfully after save, users see a generic notification ("Element creatd" / "Element updated"). You can customize this message by passing a `successMessage` prop.
```

File: packages/ra-core/src/controller/useCreateController.ts [hunk 3]
```diff
@@ -90,9 +92,13 @@ const useCreateController = (props: CreateProps): CreateControllerProps => {
                     onSuccess: onSuccess
                         ? onSuccess
                         : ({ data: newRecord }) => {
-                              notify('ra.notification.created', 'info', {
-                                  smart_count: 1,
-                              });
+                              notify(
+                                  message ? message : 'ra.notification.created',
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/useCreateController.ts]:
```suggestion
                                  message || 'ra.notification.created',
```

File: packages/ra-core/src/controller/useEditController.ts [hunk 4]
```diff
@@ -55,16 +56,21 @@ export interface EditControllerProps {
...
     const translate = useTranslate();
     const notify = useNotify();
     const redirect = useRedirect();
     const refresh = useRefresh();
     const version = useVersion();
     const { data: record, loading } = useGetOne(resource, id, {
         version, // used to force reload
-        onFailure: () => {
-            notify('ra.notification.item_doesnt_exist', 'warning');
+        onFailure: error => {
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/useEditController.ts]: I don't agree with this change. All around the application, if the `GET_ONE` call fails, we consider that it's a 404. If you change it here, it should be changed everywhere else. But I don't want to do that, so please revert it.

File: packages/ra-core/src/controller/useEditController.ts [hunk 5]
```diff
@@ -95,7 +101,7 @@ const useEditController = (props: EditProps): EditControllerProps => {
                 {
                     onSuccess: () => {
                         notify(
-                            'ra.notification.updated',
+                            message ? message : 'ra.notification.updated',
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/useEditController.ts]:
```suggestion
                            message || 'ra.notification.updated',
```

File: packages/ra-core/src/controller/useShowController.ts [hunk 6]
```diff
@@ -55,8 +55,13 @@ const useShowController = (props: ShowProps): ShowControllerProps => {
     const version = useVersion();
     const { data: record, loading } = useGetOne(resource, id, {
         version, // used to force reload
-        onFailure: () => {
-            notify('ra.notification.item_doesnt_exist', 'warning');
+        onFailure: error => {
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/useShowController.ts]: same, please revert

**Comment by natrim** [comment]
updated

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**